packages/ckb-indexer/src/indexer.ts

```diff
@@ -0,0 +1,413 @@
+import {
+  Cell,
+  CellCollector,
+  CellCollectorResults,
+  Hexadecimal,
+  HexString,
+  Indexer,
+  QueryOptions,
+  Script,
+  Tip,
+  OutPoint,
+  HexNumber,
+} from "@ckb-lumos/base";
+import { RPC } from "@ckb-lumos/rpc";
+import axios from "axios";
+
+export enum ScriptType {
+  type = "type",
+  lock = "lock",
+}
+
+export enum Order {
+  asc = "asc",
+  desc = "desc",
+}
+
+export interface CkbQueryOptions extends QueryOptions {
+  outputDataLenRange?: HexadecimalRange;
+  outputCapacityRange?: HexadecimalRange;
+  sizeLimit?: number;
+}
+
+export type HexadecimalRange = [Hexadecimal, Hexadecimal];
+export interface SearchFilter {
+  script?: Script;
+  output_data_len_range?: HexadecimalRange; //empty
+  output_capacity_range?: HexadecimalRange; //empty
+  block_range?: HexadecimalRange; //fromBlock-toBlock
+}
+export interface SearchKey {
+  script: Script;
+  script_type: ScriptType;
+  filter?: SearchFilter;
+}
+
+export interface GetLiveCellsResult {
+  last_cursor: string;
+  objects: IndexerCell[];
+}
+
+export interface rpcResponse {
+  status: number;
+  data: rpcResponseData;
+}
+
+export interface rpcResponseData {
+  result: string;
+  error: string;
+}
+
+export interface IndexerCell {
+  block_number: Hexadecimal;
+  out_point: OutPoint;
+  output: {
+    capacity: HexNumber;
+    lock: Script;
+    type?: Script;
+  };
+  output_data: HexString;
+  tx_index: Hexadecimal;
+}
+
+export interface TerminatorResult {
+  stop: boolean;
+  push: boolean;
+}
+
+export declare type Terminator = (
+  index: number,
+  cell: Cell
+) => TerminatorResult;
+
+const DefaultTerminator: Terminator = () => {
+  return { stop: false, push: true };
+};
+
+export type HexNum = string;
+export type IOType = "input" | "output";
+export type Bytes32 = string;
+export type GetTransactionsResult = {
+  block_number: HexNum;
+  io_index: HexNum;
+  io_type: IOType;
+  tx_hash: Bytes32;
+  tx_index: HexNum;
+};
+export interface GetTransactionsResults {
+  last_cursor: string;
+  objects: GetTransactionsResult[];
+}
+
+export interface GetCellsResults {
+  lastCursor: string;
+  objects: Cell[];
+}
+
+export interface AdditionalOptions {
+  sizeLimit?: number;
+  order?: Order;
+  lastCursor?: string | undefined;
+}
+
+export class CkbIndexer implements Indexer {
+  uri: string;
+
+  constructor(public ckbRpcUrl: string, public ckbIndexerUrl: string) {
```

Comment:
```suggestion
  constructor(public ckbIndexerUrl: string, public ckbRpcUrl: string) {
```

maybe we will replace the ckbIndexerUrl with Mercury endpoint in the future, once its done, the ckbRpcUrl is not necessary

Reply:
feat(ckb-indexer): change indexer constructor params position